File: MSR_TDC_tStudent/MSR_TDC_tStudent.R
rm(list = ls(all = TRUE))
# setwd('C:/...')

install.packages("lattice")
#library(lattice)

d = 3:103
r = seq(-1, 1, by = 0.02)
m = length(d)
n = length(r)
D = matrix(rep(d, each = n), nrow = n)
R = matrix(rep(r, m), nrow = n)

x = t(R)
y = t(D)
N = length(x)

t = 2 - 2 * pt(sqrt(y + 1) * sqrt(1 - x)/sqrt(1 + x), y + 1)

s = expand.grid(d = d, r = r) 

wireframe(t ~ d*d, s, shade = TRUE, xlab = "Degrees of Freedom",
    ylab = "Correlation", zlab = "Tail Dep Coef", 
    main = "Tail Dependence Coefficient for t-Student Copula", 
    scales = list(arrows = FALSE))

File: MSR_TDC_tStudent/MSR_TDC_tStudent.m

function MSR_TDC_tStudent
d     = 3:103;
r     = -1:.02:1;
[r,d] = meshgrid(r, d);
x     = r(:);
y     = d(:);
N     = length(x);
t     = 2 - 2*tcdf(sqrt(y + 1).*sqrt(1 - x)./sqrt(1 + x), y + 1);
f     = reshape(t, sqrt(N), []);

figure;
mesh(d, r, f);
axis([0 110 -1 1 0 1])
xlabel('DoF');
ylabel('rho');
zlabel('TDC');
xlabel('Degrees of Freedom');
ylabel('Correlation');
zlabel('Tail Dependence Coefficient');
title('Tail Dependence Coefficient for t-Student Copula')